script/mafia

```diff
@@ -26,16 +26,23 @@ run_upgrade () {
 
   trap clean_up EXIT
 
+  MAFIA_CUR="$0"
+
+  if [ -L "$MAFIA_CUR" ]; then
+    echo 'Refusing to overwrite a symlink; run `upgrade` from the canonical path.' >&2
```

Comment:
The C function is, but I don't believe the command is part of POSIX. I've never used a UNIX that didn't have it though, so maybe it's good enough.